awx_collection/plugins/modules/execution_environment.py

```diff
@@ -79,17 +78,21 @@ def main():
     argument_spec = dict(
         name=dict(required=True),
         new_name=dict(),
-        image=dict(required=True),
+        image=dict(),
         description=dict(),
         organization=dict(),
         credential=dict(),
         state=dict(choices=['present', 'absent', 'exists'], default='present'),
         # NOTE: Default for pull differs from API (which is blank by default)
         pull=dict(choices=['always', 'missing', 'never'], default='missing'),
     )
+    required_if = [
+        ['state', 'present', ['image']],
+        ['state', 'exists', ['image']],
+    ]
```

Comment:
I would prefer to not even have this condition. Someone could modify a different field of an existing item. If it is missing (on creation) then the API error should suffice.